### PR TITLE
[mocking] file logger

### PR DESCRIPTION
**Pull request by michalczmiel**

### CODE REVIEW

File: mocking/file-logger/go/logger_test.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+}
+
+func (s *InMemoryFileStorage) AppendStringToFile(fileName, message string) error {
+	s.Logs = append(s.Logs, message)
+
+	if s.Log == nil {
+		s.Log = make(map[string][]string)
+	}
+
+	if s.Log[fileName] == nil {
```
Comment (roxel) [inline mocking/file-logger/go/logger_test.go]: `if _, exists := s.Log[fileName]; !exists {`

File: mocking/file-logger/go/logger.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+const (
+	DefaultWeekendFileName = "weekend.txt"
+)
+
+type FileLogger struct {
+	storage FileStorage
+	clock   clock
+}
+
+func (FileLogger) formatWeekdayFile(currentTime *time.Time) string {
```
Comment (roxel) [inline mocking/file-logger/go/logger.go]: I wouldn't put so many methods unrelated to the struct. Use just functions instead

File: mocking/file-logger/go/go.mod [hunk 3]
```diff
@@ -0,0 +1,3 @@
+module logger
+
+go 1.16
```
Comment (roxel) [inline mocking/file-logger/go/go.mod]: why not use latest version, 1.19 or 1.18?

File: mocking/file-logger/go/logger.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	}
+
+	if !l.storage.FileExists(DefaultWeekendFileName) {
+		return DefaultWeekendFileName
+	}
+
+	modTime, err := l.storage.FileModificationTime(DefaultWeekendFileName)
+
+	if err != nil {
+		panic("Couldn't get the modification time from " + DefaultWeekendFileName)
```
Comment (roxel) [inline mocking/file-logger/go/logger.go]: unless in special cases, I'd just return the error up, wrapped if needed. panics are to be avoided in Go

File: mocking/file-logger/go/logger_test.go [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+	"time"
+)
+
+type InMemoryFileStorage struct {
+	Logs             map[string][]string
+	ModificationTime map[string]time.Time
+}
+
+func (s *InMemoryFileStorage) AppendStringToFile(path, message string) error {
+	if s.Logs == nil {
```
Comment (roxel) [inline mocking/file-logger/go/logger_test.go]: if you need those checks, then maybe better consider using constructor function:
```
func NewInMemoryFileStorage() *InMemoryFileStorage {
    return &InMemoryFileStorage{Logs: make....
}
```

File: mocking/file-logger/go/storage.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"errors"
```
Comment (roxel) [inline mocking/file-logger/go/storage.go:4]: different styling: using tabs whereas in other places double spaces

File: mocking/file-logger/go/storage.go [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+		return err
+	}
+
+	return nil
+}
+
+func (localFileStorage) FileExists(path string) bool {
+	_, err := os.Stat(path)
+
+	if errors.Is(err, os.ErrNotExist) {
```
Comment (roxel) [inline mocking/file-logger/go/storage.go]: consider all such cases:
you could just write:
```
return errors.Is(err, os.ErrNotExist)
```

File: mocking/file-logger/go/storage.go [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+		return false
+	}
+
+	return true
+}
+
+func (localFileStorage) RenameFile(oldPath, newPath string) error {
+	err := os.Rename(oldPath, newPath)
```
Comment (roxel) [inline mocking/file-logger/go/storage.go]: and here `return os.Rename(oldPath, newPath)`

File: mocking/file-logger/go/storage.go (hunk 6 above)
Comment (michalczmiel) [reply mocking/file-logger/go/storage.go:4]: I don't know why it's formatted like that. I have used editorconfig and go fmt

File: mocking/file-logger/go/logger.go [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+		previousSaturday = &saturday
+	}
+
+	newFileName := formatPreviousWeekendFile(previousSaturday)
+
+	// rename the old file created in past weekend
+	err = l.storage.RenameFile(DefaultWeekendFileName, newFileName)
+	if err != nil {
+		fmt.Printf("Couldn't rename old %v", DefaultWeekendFileName)
+		return err, ""
```
Comment (roxel) [inline mocking/file-logger/go/logger.go]: maybe `return fmt.Errorf("Couldn't rename old %v. error: %v", DefaultWeekendFileName, err)`

File: mocking/file-logger/go/logger.go [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+	// check if file was created during this weekend
+	if modTime.After(currentWeekBeginning) {
+		return true
+	}
+
+	return false
+}
+
+func (l *FileLogger) getFileName() (error, string) {
```
Comment (roxel) [inline mocking/file-logger/go/logger.go]: convention is to always return error in last value so `(string, error)`

**Pull request merged**